go-controller/pkg/retry/obj_retry.go

```diff
@@ -492,6 +493,9 @@ func (r *RetryFramework) WatchResourceFiltered(namespaceForFilteredHandler strin
 		labelSelectorForFilteredHandler, // filter out objects not matching these labels
 		cache.ResourceEventHandlerFuncs{
 			AddFunc: func(obj interface{}) {
+				if !r.ResourceHandler.FilterResource(obj) {
```

Comment:
what is the meaning of "Filter Resource"? I would think filtering means to "filter it out" and not process it, but these checks seem to have the inverse behavior. If we do not filter it, then we return?

Reply:
I agree this is confusing, I have changed it to `FilterOutResource` that returns true to exclude an object.